x-pack/libbeat/management/managerV2.go

```diff
@@ -613,7 +636,7 @@ func (cm *BeatV2Manager) reloadInputs(inputUnits []*client.Unit) error {
 }
 
 // this function is registered as a debug hook
-// it prints the last known configuration genreated by the beat
+// it prints the last known configuration generated by the beat
```

Comment:
I think you will want to save the most recently received feature configuration and dump it out in this function. It can follow the same pattern the output configuration uses of saving it and then writing it out.

This function generates a file called `beat-rendered-config.yml` in the components sub-directories for Beats in the agent diagnostics, so you can inspect the Beat configuration that was generated based on the agent policy configuration. This is really valuable for debugging so we'll want to keep it up to date.

You will probably want to add a Features entry to the file structure. For an example of what this file looks like run `elastic-agent diagnostics`, unpack the archive, and search for the `beat-rendered-config.yml`. I can help you interpret the diagnostics if it isn't obvious what the folder structure is, the top level is documented in https://www.elastic.co/guide/en/fleet/current/elastic-agent-cmd-options.html#elastic-agent-diagnostics-command

https://github.com/elastic/beats/blob/e7e6dacfca2055d14fba269253dade2d197906d8/x-pack/libbeat/management/managerV2.go#L642-L650

Reply:
This is really helpful context, thank you, @cmacknz! I will make sure `beat-rendered-config.yml` contains feature flag information.

Reply:
This is implemented now.

<img width="1642" alt="Screenshot 2023-03-16 at 07 13 05" src="https://user-images.githubusercontent.com/51061/225600108-6366ad12-086c-40f6-8193-e7869b9b4895.png">